### ISSUE TITLE
Missing warning for local function not referenced outside of itself

### ISSUE THREAD

**Issue by jcouv** [dotnet/roslyn#31707]
I expected a warning for this:
```C#
public class C 
{
    public void M() 
    {
        void local() // expecting warning CS8321: The local function 'local' is declared but never used
        {
            local();
        }
    }
}
```

@agocke What do you think? Is this correct? Is this worth tagging for consideration in warning waves?

**Comment by agocke** [comment]
That is incorrect. Our "unused" warnings are entirely tree-walker based, not flow analysis based. You can check this by using a variable in unreachable code and seeing that the "unused" warning goes away: https://sharplab.io/#v2:CYLg1APgAgDABFAjAbgLACgoGYECY4DCcA3hnOQjlACxwCyAFAJQlkXtvvk1wAyzJOAF9OXUeygB2NOi4V+TGexHohQA

**Comment by jcouv** [comment]
Ok. Closing as by-design then. Thanks
